src/liballoc/boxed.rs

```diff
@@ -511,3 +512,53 @@ impl<'a,A,R> FnOnce<A> for Box<FnBox<A,Output=R>+Send+'a> {
 }
 
 impl<T: ?Sized+Unsize<U>, U: ?Sized> CoerceUnsized<Box<U>> for Box<T> {}
+
+impl<T: Clone> Clone for Box<[T]> {
+    fn clone(&self) -> Self {
+        let mut new = BoxBuilder {
+            data: RawVec::with_capacity(self.len()),
+            len: 0
+        };
+
+        let mut target = new.data.ptr();
+
+        for item in self.iter() {
+            unsafe {
+                ptr::write(target, item.clone());
+                target = target.offset(1);
+            };
+
+            new.len += 1;
+        }
+
+        unsafe { new.into_box() }
+    }
+}
+
+struct BoxBuilder<T> {
+    data: RawVec<T>,
+    len: usize,
+}
+
+impl<T> BoxBuilder<T> {
+    unsafe fn into_box(self) -> Box<[T]> {
+        let raw = ptr::read(&self.data);
+        mem::forget(self);
+        raw.into_box()
+    }
```

Comment:
Honestly now that you mention it, I'd even hoist the type into the function so no one else can use it.